baselines/jft/active_learning.py

```diff
@@ -0,0 +1,450 @@
+
+    # Take care of the mask.
+    entropy = jnp.where(mask, entropy, NINF_SCORE)
+
+    # TODO: if we want to run on multi host then this requires jax.lax.all_gather(entropy, axis_name='batch')
+    return entropy
+
+  entropies = []
+  ids = []
+  for batch in iter_ds:
+    batch_entropy = compute_batch_entropy(opt_repl.target, batch["image"], batch["mask"])
+    # TODO: if we run on multi host, this needs to be used as batch_entropy[0]
+
+    flat_batch_entropy = np.array(batch_entropy).ravel()
+    flat_id = np.array(batch["id"]).ravel()
+    assert flat_batch_entropy.shape == flat_id.shape
+    entropies.append(flat_batch_entropy)
+    ids.append(flat_id)
+
+  entropies = np.concatenate(entropies)
+  ids = np.concatenate(ids)
+
+  # NOTE: due to batch padding, entropies/ids will be of size:
+  # if training set size % batch size > 0:
+  # (training set size // batch size + 1) * batch size
+  # else:
+  # just training set size
+
+  return ids, entropies
+
+
+def select_acquisition_batch_indices(
+    *, model, opt_repl, pool_ds, acquisition_batch_size, ignored_ids, prefetch_to_device=1
+):
+  iter_ds = input_utils.start_input_pipeline(pool_ds, prefetch_to_device)
+  ids, scores = get_id_entropy(model=model, opt_repl=opt_repl, iter_ds=iter_ds)
+
+  # Ignore already selected ids
+  # TODO: vectorize this
+  ids_list = ids.tolist()
+  for ignored_id in ignored_ids:
+    scores[ids_list.index(ignored_id)] = NINF_SCORE
+
+  f_ent = scores[scores > 0]
+  print(f"Score statistics pool set - min: {f_ent.min()}, mean: {f_ent.mean()}, max: {f_ent.max()}")
+
+  top_scorers = np.argpartition(-scores, acquisition_batch_size)[:acquisition_batch_size]
+
+  top_ids = ids[top_scorers].tolist()
+  top_scores = scores[top_scorers].tolist()
+
+
+  print(f"Data selected - ids: {top_ids}, with scores: {top_scores}")
+
+  return top_ids, top_scores
+
+
+def get_accuracy(*, evaluation_fn, opt_repl, ds, prefetch_to_device=1):
+  iter_ds = input_utils.start_input_pipeline(ds, prefetch_to_device)
+
+  ncorrect, nseen = [], []
+  for batch in iter_ds:
+    batch_ncorrect, batch_losses, batch_n, batch_metric_args = evaluation_fn(
+      opt_repl.target, batch["image"], batch["labels"], batch["mask"]
+    )
+
+    ncorrect += [batch_ncorrect[0]]
+    nseen += [batch_n[0]]
+
+  ncorrect = np.sum(ncorrect)
+  nseen = np.sum(nseen)
+
+  return ncorrect / nseen
+
+
+def finetune(*, update_fn, opt_repl, lr_fn, ds, rngs_loop, total_steps, prefetch_to_device=1):
+  iter_ds = input_utils.start_input_pipeline(ds, prefetch_to_device)
+  lr_iter = train_utils.prefetch_scalar(
+    map(lr_fn, range(total_steps)), prefetch_to_device
+  )
+
+  for step, train_batch, lr_repl in zip(tqdm.trange(1, total_steps + 1), iter_ds, lr_iter):
+    opt_repl, l, rngs_loop, measurements = update_fn(opt_repl, lr_repl, train_batch["image"], train_batch["labels"], rngs_loop)
+
+  return opt_repl, rngs_loop
+
+
+def make_init(model, test_ds, local_batch_size, config):
```

Comment:
nit: for consistency with the other `make_*` functions, it might be nice to name this `make_init_fn`.